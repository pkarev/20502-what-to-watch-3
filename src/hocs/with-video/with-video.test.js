import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withVideo from './with-video.jsx';

const MockComponent = ({renderVideo}) => (
  <div>
    {renderVideo({
      src: `/test/src`,
      poster: `/test/poster`,
    })}
  </div>
);

MockComponent.propTypes = {
  renderVideo: PropTypes.func.isRequired,
};

const MockComponentWithVideo = withVideo(MockComponent);
it(`Component with video`, () => {
  const tree = renderer
    .create(
        <MockComponentWithVideo/>, {
          createNodeMock: () => ({}),
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
