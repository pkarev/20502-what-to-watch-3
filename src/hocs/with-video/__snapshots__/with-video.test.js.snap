// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component with video 1`] = `
<div>
  <video
    autoPlay={false}
    controls={false}
    muted={false}
    poster="/test/poster"
    style={
      Object {
        "height": "100%",
        "objectFit": "cover",
        "objectPosition": "center center",
        "width": "100%",
      }
    }
  >
    <source
      src="/test/src"
    />
  </video>
</div>
`;
